integration-tests/src/tests/client/features/increase_storage_compute_cost.rs

```diff
@@ -0,0 +1,369 @@
+    // It should not be affected by compute costs, as it doesn't access storage.
+    let method_name = "sum_n".to_owned();
+    // note: If we make `method_args` only 1M, then the gas cost increase due to
+    // finite-wasm change (also included in version 61) is large enough that we
+    // see the number of receipts fitting in a chunk going from 64 to 61 even
+    // without the compute costs. 10.03M is the sweet spot where the granularity is
+    // course enough that the finite-wasm change doesn't interfere. We can
+    // process exactly 8 receipts per chunk before and after finite-wasm.
+    // And if compute costs would additionally affect this, the test would fail
+    // because it is very close at the border between 7 or 8 receipts fitting
+    // (7 receipts with finite wasm require 999 Tgas, 8 receipts require 1142 Tgas).
+    let method_args: Vec<u8> = 10_030_000u64.to_le_bytes().to_vec();
+    let num_transactions = 20;
+    let uses_storage = false;
+    let fails: bool = false;
+    assert_compute_limit_reached(method_name, method_args, num_transactions, uses_storage, fails);
+}
+
+/// Test the case where a function call fails and the limit is unaffected by compute costs.
+#[test]
+fn test_non_storage_gas_exceeded() {
+    // `loop_forever()` loops until either gas is exhausted.
+    // It should not be affected by compute costs, as it doesn't access storage.
+    let method_name = "loop_forever".to_owned();
+    let method_args: Vec<u8> = vec![];
+    let num_transactions = 20;
+    let uses_storage = false;
+    let fails = true;
+    assert_compute_limit_reached(method_name, method_args, num_transactions, uses_storage, fails);
+}
+
+/// Checks that a specific function call reaches the expected limits.
+///
+/// (This is a helper function called by all tests above)
+///
+/// The function call is on the test contract
+/// (`near_test_contracts::rs_contract()`). The gas limit is tested at version
+/// 60 (before compute costs) and the limit is checked again with version 61
+/// (with compute costs). The boolean `uses_storage` defines
+/// whether we expect the second limit to be more restrictive or if they should
+/// both fill a chunk equally.
+///
+/// Regarding `num_transactions`, this is how many function calls are queued up,
+/// It should not more than can be converted in a single chunk but more than can
+/// be executed in a single chunk. Otherwise, the test doesn't work to check the
+/// limits and consequently some assertions will fail.
+fn assert_compute_limit_reached(
+    method_name: String,
+    method_args: Vec<u8>,
+    num_transactions: u64,
+    uses_storage: bool,
+    should_fail: bool,
+) {
+    // The immediate protocol upgrade needs to be set for this test to pass in
+    // the release branch where the protocol upgrade date is set.
+    std::env::set_var("NEAR_TESTS_IMMEDIATE_PROTOCOL_UPGRADE", "1");
+    near_o11y::testonly::init_test_logger();
+
+    let new_protocol_version = 61;
```

Comment:
```suggestion
    let new_protocol_version = ProtocolFeature::ComputeCosts.protocol_version();
```

Reply:
I don't think this is accurate. `ProtocolFeature::ComputeCosts.protocol_version()` is when we introduce NEP-455. It's just a "coincident" that we also change storage related costs in the same version. But we could also do storage cost changes in version 62.

Reply:
Hmmmm. Then I would introduce separate protocol feature like `StorageComputeCosts` for consistency.
And why we don't have specific test for `ProtocolFeature::ComputeCosts` then?

Reply:
Have we done that for gas price changes in the past? Because the change is defined in 61.yaml and we can't do `cfg` for that. So it would be a completely dead enum variant that's used nowhere and therefore doesn't turn on/off anything. And the assignment in `version.rs` to version 61 would make no difference at all. That's a general problem for changes that are defined by the config and not by code, like this change here.

Regarding tests, we only have tests in https://github.com/near/nearcore/blob/master/runtime/near-vm-logic/src/gas_counter.rs#L356. These are not what I consider integrations tests, though. They just check the the compute limit is as expected.

When we merged that, I felt that a integration tests as done now isn't strictly necessary, because we don't expect anything to be observable with the compute costs feature alone. Whereas now, when we set the compute costs, we know the end-to-end effect we expect and can write an integration test for it.

But now I am not sure about that reasoning. Do you think it would make sense in the future to always require integration tests when we add a protocol feature?

Reply:
> Have we done that for gas price changes in the past?

We did it for `LowerStorageCost`, `LowerDataReceiptAndEcrecoverBaseCost`, `LowerRegularOpCost` and some more. I think it's a good idea to have indication of such protocol changes in the code.

Compute costs are a cornercase because they didn't change behaviour alone. And 61 in the code leaves a strong feeling that it should be tied with `ProtocolFeature::ComputeCosts`.

Some weird compromise I came up with:
```
    const INCREASED_STORAGE_COSTS_PROTOCOL_VERSION = 61;
    let new_protocol_version = INCREASED_STORAGE_COSTS_PROTOCOL_VERSION;
	assert!(new_protocol_version >= ProtocolFeature::ComputeCosts.protocol_version());
    let old_protocol_version = new_protocol_version - 1;
    ...
```

Reply:
> We did it for LowerStorageCost, LowerDataReceiptAndEcrecoverBaseCost, LowerRegularOpCost and some more. I think it's a good idea to have indication of such protocol changes in the code.

Oh right, I didn't know that or maybe just forgot. Seems weird to me to have a feature that doesn't actually do anything.

I like that compromise! I can see this is still quite hacky... But it does reflect the (somewhat ugly) reality that we have a protocol feature that could be turned on/off independently of what we have in 61.yaml, but that would cause problems. So having an assertion instead of just assigning `ProtocolFeature::ComputeCosts.protocol_version()` makes sense to me.

Reply:
Everything seems to be resolved, I'll add the merge label